pkg/command/output.go

```diff
@@ -40,46 +43,61 @@ func AddJSONOutput(cmd *cobra.Command) {
 //PrintOutput receives an interface and dump the data using the --output flag.
 //ATM only json or jsonpath. In the future yaml
 func PrintOutput(data interface{}) error {
-	var re = regexp.MustCompile(`^jsonpath\=(.*)`)
+	return PrintOutputWithType(data, outputOpt)
+}
 
-	if outputOpt == "json" {
+//PrintOutputWithType receives an interface and dump the data using the --output flag.
+//ATM only json or jsonpath. In the future yaml
+func PrintOutputWithType(data interface{}, outputType string) error {
+	if outputType == "json" {
 		return dumpJSON(data, "")
 	}
 
-	if re.MatchString(outputOpt) {
-		return dumpJSON(data, re.ReplaceAllString(outputOpt, "$1"))
+	if re.MatchString(outputType) {
+		return dumpJSON(data, re.ReplaceAllString(outputType, "$1"))
 	}
 
 	return fmt.Errorf("Couldn't found output printer")
 }
 
-// dumpJSON dump the data variable to the stdout as json.
-// If somethings fail, it'll return an error
-// If jsonPath is passed, it'll run the json query over data var.
-func dumpJSON(data interface{}, jsonPath string) error {
-
+// DumpJSONToString dumps the contents of data into a string. If jsonpath
+// is non-empty, will attempt to do jsonpath filtering using said string.
+// Returns the string representation of the JSON in data, or an error if
+// any JSON marshaling, parsing operations fail.
+func DumpJSONToString(data interface{}, jsonPath string) ([]byte, error) {
```

Comment:
function description and name don't describe the real behavior of the function.

Reply:
Nice catch, fixed.